modules/stochastic_tools/include/samplers/AdaptiveImportanceSamplerActiveLearning.h

```diff
@@ -0,0 +1,25 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "AdaptiveImportanceSampler.h"
+#include "ReporterInterface.h"
+
+/**
+ * A class used to perform Adaptive Importance Sampling using a Markov Chain Monte Carlo algorithm
+ * and Gaussian Process active learning
+ */
+class AdaptiveImportanceSamplerActiveLearning : public AdaptiveImportanceSampler
+{
+public:
+  static InputParameters validParams();
+
+  AdaptiveImportanceSamplerActiveLearning(const InputParameters & parameters);
+};
```

Comment:
If the difference is only one thing, could you try to unify the regular and AL behavior? I don't see a reason why we should create an almost empty class for AIS-AL at this point.

Reply:
You're right. I wanted to have a separate class for AIS with AL for two main reasons: (1) expansion of the Markov chain proposals during active learning if interested in the future; and (2) easier documentation and understanding.